src/H5FDs3comms.c

```diff
@@ -753,13 +757,15 @@ H5FD__s3comms_s3r_getsize(s3r_t *handle)
  *
  *              fa can be NULL (implies no authentication)
  *              fapl_token can be NULL
+ *              fapl_endpoint can be NULL
  *
  * Return:      SUCCESS:    Pointer to new request handle.
  *              FAILURE:    NULL
  *----------------------------------------------------------------------------
  */
 s3r_t *
-H5FD__s3comms_s3r_open(const char *url, const H5FD_ros3_fapl_t *fa, const char *fapl_token)
+H5FD__s3comms_s3r_open(const char *url, const H5FD_ros3_fapl_t *fa, const char *fapl_token,
```

Comment:
This seems a bit odd to add an extra parameter for an endpoint that is generally just going to be `NULL` when it should just be what's passed in `url` in the first place.